ServerCore/Helpers/PlayerEventHelper.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Identity;
+using ServerCore.DataModel;
+
+namespace ServerCore.Helpers
+{
+    public class PlayerEventHelper
+    {
+        /// <summary>
+        /// Returns a list of the author's puzzles for the given event
+        /// </summary>
+        /// <param name="thisEvent">The event that's being checked</param>
+        /// <param name="puzzleServerContext">Current PuzzleServerContext</param>
+        public List<Puzzle> GetPuzzlesForAuthorAndEvent(PuzzleServerContext dbContext, Event thisEvent, PuzzleUser user)
+        {
+            return dbContext.PuzzleAuthors.Where(pa => pa.Author.ID == user.ID && pa.Puzzle.Event.ID == thisEvent.ID).Select(pa => pa.Puzzle).ToList();
```

Comment:
It might be better to leave this as a query for better perf - just skip the ToList() part. Then a caller can call ToListAsync() at the proper time. If not, at least use ToListAsync().